autopush_rs/src/server/mod.rs

```diff
@@ -485,6 +504,29 @@ impl Server {
         Ok((srv2, core))
     }
 
+    /// Create an v1 or v2 WebPush endpoint from the identifiers
+    ///
+    /// Both endpoints use bytes instead of hex to reduce ID length.
+    //  v1 is the uaid + chid
+    //  v2 is the uaid + chid + sha256(key).bytes
+    pub fn make_endpoint(&self, uaid: &Uuid, chid: &Uuid, key: Option<String>) -> Result<String> {
+        let root = format!("{}/wpush/", self.opts.endpoint_url);
+        let mut base = hex::decode(uaid.simple().to_string()).chain_err(|| "Error decoding")?;
+        base.extend(hex::decode(chid.simple().to_string()).chain_err(|| "Error decoding")?);
+        if let Some(k) = key {
+            let raw_key = base64::decode_config(&k, base64::URL_SAFE)
+                .chain_err(|| "Error encrypting payload")?;
+            let key_digest = hash::hash(hash::MessageDigest::sha256(), &raw_key)
+                .chain_err(|| "Error creating message digest for key")?;
+            base.extend(key_digest.iter());
+            let encrypted = self.opts.fernet.encrypt(&base);
```

Comment:
let's trim_matches('=') the result of fernet.encrypt here and below (like the python code), just to be super safe

Reply:
The Python one only strips the encrypted value once.

Reply:
I meant below as in the else block